diff -rpu -N orig/configure ffmpeg-mt/configure
--- orig/configure	2010-07-27 13:05:17.865562700 -0700
+++ ffmpeg-mt/configure	2010-07-27 13:05:29.670743100 -0700
@@ -1788,7 +1788,7 @@ if   $cc -v 2>&1 | grep -qi ^gcc; then
         CC_DEPFLAGS='-MMD -MF $(@:.o=.d) -MT $@'
         AS_DEPFLAGS='-MMD -MF $(@:.o=.d) -MT $@'
     fi
-    speed_cflags='-O3'
+    speed_cflags='-O2'
     size_cflags='-Os'
 elif $cc --version 2>/dev/null | grep -q Intel; then
     cc_type=icc
@@ -2607,6 +2607,9 @@ if ! disabled pthreads && ! enabled w32t
         add_extralibs -pthreads
     elif check_func pthread_create -lpthreadGC2; then
         add_extralibs -lpthreadGC2
+    elif check_func pthread_create -lpthreadGC2 -lws2_32; then
+        add_cflags -DPTW32_STATIC_LIB
+        add_extralibs -lpthreadGC2 -lws2_32
     elif ! check_lib pthread.h pthread_create -lpthread; then
         disable pthreads
     fi
@@ -2823,7 +2826,6 @@ elif enabled ccc; then
 elif enabled gcc; then
     check_cflags -fno-tree-vectorize
     check_cflags -Werror=implicit-function-declaration
-    check_cflags -Werror=missing-prototypes
 elif enabled clang; then
     check_cflags -Qunused-arguments
 elif enabled armcc; then
diff -rpu -N orig/libavcodec/allcodecs.c ffmpeg-mt/libavcodec/allcodecs.c
--- orig/libavcodec/allcodecs.c	2010-07-27 13:05:18.281604300 -0700
+++ ffmpeg-mt/libavcodec/allcodecs.c	2010-07-27 13:05:29.675743600 -0700
@@ -45,6 +45,14 @@
           extern AVBitStreamFilter x##_bsf; \
           if(CONFIG_##X##_BSF)     av_register_bitstream_filter(&x##_bsf); }
 
+#ifdef PTW32_STATIC_LIB
+static void detach_ptw32(void)
+{
+    pthread_win32_thread_detach_np();
+    pthread_win32_process_detach_np();
+}
+#endif
+
 void avcodec_register_all(void)
 {
     static int initialized;
@@ -53,6 +61,11 @@ void avcodec_register_all(void)
         return;
     initialized = 1;
 
+#ifdef PTW32_STATIC_LIB
+    pthread_win32_process_attach_np();
+    pthread_win32_thread_attach_np();
+    atexit(detach_ptw32);
+#endif
     /* hardware accelerators */
     REGISTER_HWACCEL (H263_VAAPI, h263_vaapi);
     REGISTER_HWACCEL (H264_DXVA2, h264_dxva2);
