diff -rpu -N orig/libavformat/mov.c ffmpeg-mt/libavformat/mov.c
--- orig/libavformat/mov.c	2010-07-27 13:06:22.069982500 -0700
+++ ffmpeg-mt/libavformat/mov.c	2010-07-27 13:06:23.245100000 -0700
@@ -344,6 +344,8 @@ static int mov_read_dref(MOVContext *c,
         MOVDref *dref = &sc->drefs[i];
         uint32_t size = get_be32(pb);
         int64_t next = url_ftell(pb) + size - 4;
+        if (size < 8)
+            return -1;
 
         dref->type = get_le32(pb);
         get_be32(pb); // version + flags
