Only in unpatched_ffmpeg-mt: ffmpeg-mt
diff -rpu unpatched_ffmpeg-mt/libavcodec/Makefile ffmpeg-mt/libavcodec/Makefile
--- unpatched_ffmpeg-mt/libavcodec/Makefile	2010-03-11 17:46:30 -0800
+++ ffmpeg-mt/libavcodec/Makefile	2010-03-11 19:12:33 -0800
@@ -493,8 +493,7 @@ OBJS-$(CONFIG_MATROSKA_MUXER)          +
 OBJS-$(CONFIG_MOV_DEMUXER)             += mpeg4audio.o mpegaudiodata.o
 OBJS-$(CONFIG_MPEGTS_MUXER)            += mpegvideo.o
 OBJS-$(CONFIG_NUT_MUXER)               += mpegaudiodata.o
-OBJS-$(CONFIG_OGG_DEMUXER)             += flacdec.o flacdata.o flac.o \
-                                          dirac.o mpeg12data.o
+OBJS-$(CONFIG_OGG_DEMUXER)             += dirac.o mpeg12data.o
 OBJS-$(CONFIG_OGG_MUXER)               += xiph.o flacdec.o flacdata.o flac.o
 OBJS-$(CONFIG_RTP_MUXER)               += mpegvideo.o

diff -rpu unpatched_ffmpeg-mt/libavformat/oggdec.c ffmpeg-mt/libavformat/oggdec.c
--- unpatched_ffmpeg-mt/libavformat/oggdec.c	2010-03-11 17:46:31 -0800
+++ ffmpeg-mt/libavformat/oggdec.c	2010-03-11 19:12:33 -0800
@@ -38,13 +38,23 @@
 #define DECODER_BUFFER_SIZE MAX_PAGE_SIZE

 static const struct ogg_codec * const ogg_codecs[] = {
+#if CONFIG_DIRAC_DEMUXER
     &ff_dirac_codec,
+#endif
+#if CONFIG_LIBSPEEX
     &ff_speex_codec,
+#endif
     &ff_vorbis_codec,
     &ff_theora_codec,
+#if CONFIG_FLAC_DECODER
     &ff_flac_codec,
+#endif
+#if CONFIG_DIRAC_DEMUXER
     &ff_old_dirac_codec,
+#endif
+#if CONFIG_FLAC_DECODER
     &ff_old_flac_codec,
+#endif
     &ff_ogm_video_codec,
     &ff_ogm_audio_codec,
     &ff_ogm_text_codec,
